### PR TITLE
feat: support swa

### PR DESCRIPTION
**Pull request by jimoosciuc**
FIXES: #267  and support [gemma2 ](https://arxiv.org/pdf/2408.00118)

### Launch Server
```
python3 -u -m sgl_jax.launch_server --model-path google/gemma-2-9b-it --trust-remote-code  --tp-size=4 --device=tpu --mem-fraction-static=0.8 --max-running-requests 256 --skip-server-warmup --page-size=1 --context-length=4096 --swa-full-tokens-ratio 0.2
```
### Accuracy
gemma-2-9b-it
```
evalscope eval  --model google/gemma-2-9b-it --api-url http://127.0.0.1:30000/v1/chat/completions --api-key EMPTY --eval-type service --datasets gsm8k --eval-batch-size 256
```
result:
```
modelscope/gsm8k report table:
+---------------+-----------+-----------------+----------+-------+---------+---------+
| Model         | Dataset   | Metric          | Subset   |   Num |   Score | Cat.0   |
+===============+===========+=================+==========+=======+=========+=========+
| gemma-2-9b-it | gsm8k     | AverageAccuracy | main     |  1319 |  0.8218 | default |
+---------------+-----------+-----------------+----------+-------+---------+---------+

2025-10-31 06:06:14,457 - evalscope - INFO - Skipping report analysis (`analysis_report=False`).
```

### CODE REVIEW

File: python/sgl_jax/srt/models/qwen2.py [hunk 1]
```diff
@@ -373,7 +373,7 @@ def _create_qwen2_weight_mappings(self) -> dict:
 
     def _create_layer_mappings(self, layer_idx: int) -> dict:
         prefix = f"model.layers.{layer_idx}"
-        target_prefix = f"transformer.layers.{layer_idx}"
+        target_prefix = f"model.layers.{layer_idx}"
```
Comment (aolemila) [inline python/sgl_jax/srt/models/qwen2.py:376]: It looks like the field has been changed, please check whether mapping should be modified in Tunix or not.  @pathfinder-pf

Comment (jimoosciuc) [reply python/sgl_jax/srt/models/qwen2.py:376]: Yes, the model naming implementation references the model.safetensors.index.json file for each model on HF, maintaining consistency as much as possible.

File: python/sgl_jax/srt/configs/model_config.py [hunk 2]
```diff
@@ -75,7 +75,7 @@ def __init__(
         )
 
         self.hf_text_config = get_hf_text_config(self.hf_config)
-        self.attention_chunk_size = getattr(self.hf_text_config, "attention_chunk_size", None)
```
Comment (Prayer3th) [inline python/sgl_jax/srt/configs/model_config.py:78]: Removing `attention_chunk_size` here—could this reduce compatibility with other model config.json files?

File: python/sgl_jax/srt/layers/layernorm.py [hunk 3]
```diff
@@ -171,3 +171,36 @@ def _normalize(
         args.append(bias)
     dtype = dtypes.canonicalize_dtype(*args, dtype=dtype)
     return jnp.asarray(y, dtype)
+
+
+class GemmaRMSNorm(nnx.Module):
```
Comment (Prayer3th) [inline python/sgl_jax/srt/layers/layernorm.py]: GemmaRMSNorm did not call the maybe_distributed_mean function. When tp hidden_dim was partitioned, did it only compute the local variance?

File: python/sgl_jax/srt/server_args.py [hunk 4]
```diff
@@ -753,9 +730,9 @@ def add_cli_args(parser: argparse.ArgumentParser):
             help="Set the list of batch sizes buckets for jax jit",
         )
         parser.add_argument(
-            "--disable-jax-precompile",
+            "--disable-precompile",
```
Comment (Prayer3th) [inline python/sgl_jax/srt/server_args.py:733]: Should the documentation be updated accordingly to reflect these changes in startup parameter naming?

File: python/sgl_jax/srt/models/qwen2.py (hunk 1 above)
Comment (pathfinder-pf) [reply python/sgl_jax/srt/models/qwen2.py:376]: > Yes, the model naming implementation references the model.safetensors.index.json file for each model on HF, maintaining consistency as much as possible.

please modify tunix in https://github.com/google/tunix/blob/main/tunix/models/qwen2/mapping_sglang_jax.py if the mapping change.

Comment (jimoosciuc) [reply python/sgl_jax/srt/models/qwen2.py:376]: @pathfinder-pf https://github.com/google/tunix/pull/664 Could you please help review this?

File: python/sgl_jax/srt/configs/model_config.py (hunk 2 above)
Comment (jimoosciuc) [reply python/sgl_jax/srt/configs/model_config.py:78]: attention_chunk_size is a llama4 feature that is not yet supported in our code

File: python/sgl_jax/srt/server_args.py (hunk 4 above)
Comment (jimoosciuc) [reply python/sgl_jax/srt/server_args.py:733]: DONE
